readme: testing commit for Mergify

0xc0170 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

Testing mergify bot.

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

0xc0170 committed:
readme: testing commit

fkjagodzinski reviewed (changes requested):
test

fkjagodzinski reviewed (changes requested):
Requesting changes again.

fkjagodzinski reviewed (changes requested):
Need more changes

bulislaw reviewed (dismissed):
Please change something!

0xc0170 committed:
mergify: fix label do not merge, fixes failure with action

0xc0170 committed:
mergify: add note about write access for proper review request

0xc0170 committed:
mergify: fix labels with spaces

0xc0170 committed:
mergify: to go to CI, work not matters if approved

As pr-head is neutral, means there was a push, we should test PR again and if all passed,
should be marked as ready for merge.

0xc0170 committed:
mergify: add from work to CI stage

0xc0170 committed:
mergify: also remove needs: work if CI ready

There are two cases here: either PR was in  review or needs work. in any case, new
CI should be started as pr-head is yellow. Once it completes, based on reviews, we can move
to ready for merge or back to needs work if fails.

fkjagodzinski reviewed (approved):
Hold tight! Approving now.

bulislaw reviewed (approved):
LGTM

0xc0170 committed:
mergify: simplify needs CI transition

from review or work - approval and update (pr head yellow) should result in CI run. These 2 preconditions should be
sufficient to move to CI stage.

0xc0170 committed:
mergify: instead of neutral, use not success neither failure

0xc0170 committed:
mergify: fix jenkins check if smart tester skips pipeline
